### ISSUE TITLE
What is the meaning of refpoint_embed?

### ISSUE THREAD

**Issue by Yzichen** [ZrrSkywalker/MonoDETR#39]
self.refpoint_embed = nn.Embedding(num_queries * group_num, 6)

May I ask what these 6 dimensions mean?

**Comment by Yzichen** [comment]
Thanks for your reply, is this design from DAB-DETR?

**Comment by ZrrSkywalker** [comment]
The (l, r, t, b) is especially for monocular 3D object detection adopted by [MonoFlex](https://openaccess.thecvf.com/content/CVPR2021/papers/Zhang_Objects_Are_Different_Flexible_Monocular_3D_Object_Detection_CVPR_2021_paper.pdf), since the projected 3D center may not locate at the center of the 2D box. For 2D DETR, just (w, h) is enough.

**Comment by ysyf293** [comment]
May I ask what these 'group_num' mean?

**Comment by ZrrSkywalker** [comment]
@ysyf293 We refer to the tricks in GroupDETR to utilize multiple groups for queries, 11 by default, for more stable performance.

**Comment by ysyf293** [comment]
Can you ask me the principle of this design? And why not during the testing phase? Also, I used your published code training test to submit results on kitii only 12.34, do you know what the problem is? I use a single card A6000 training.

--------------原始邮件--------------
发件人："Renrui&nbsp;Zhang ***@***.***&gt;;
发送时间：2023年9月7日(星期四) 中午11:37
收件人："ZrrSkywalker/MonoDETR" ***@***.***&gt;;
抄送："杨凡 ***@***.***&gt;;"Mention ***@***.***&gt;;
主题：Re: [ZrrSkywalker/MonoDETR] What is the meaning of refpoint_embed? (Issue #39)
-----------------------------------

 


@ysyf293 We refer to the tricks in GroupDETR to utilize multiple groups for queries, 11 by default, for more stable performance.我们参考GroupDETR中的技巧，利用多个组进行查询（默认情况下为11个），以获得更稳定的性能。

—和mdash

**Comment by ysyf293** [comment]
> @ysyf293 We refer to the tricks in GroupDETR to utilize multiple groups for queries, 11 by default, for more stable performance.

May I ask the principle of this design? And why is it not needed during the validation phase ?Also, I used your published code training test to submit results on kitii only 12.34, do you know what the problem is? I use a single card A6000 training.

**Comment by ysyf293** [comment]
I would like to ask which parameters in the setting should be selected Ture to obtain the best effect? For example, two_stage: False; use_dab: Ture; use_dn: Ture; two_stage_dino: False; TWO_STAGE_DINO: false; init_box: Ture?

**Comment by ZrrSkywalker** [comment]
The following configs can achieve the best effect. We are still investigating how the two-stage/dab/dn/dino tricks can be used to improve MonoDETR.
two_stage: False
use_dab: False
use_dn: False
two_stage_dino: False
init_box: False

**Comment by ysyf293** [comment]
Thank you very much for your reply. But why am I using the full "False" parameter Settings you mentioned, training on all trainval datasets, and submitting on kitti's website with only 12.34 accuracy?

**Comment by ZrrSkywalker** [comment]
The current code with configurations is only for KITTI val set. We will release the test code soon. Thanks for your waiting.

**Comment by yjy4231** [comment]
> The current code with configurations is only for KITTI val set. We will release the test code soon. Thanks for your waiting.

![image](https://github.com/ZrrSkywalker/MonoDETR/assets/52461074/ec2347f0-3f3a-4223-81c1-d89c73c9b91d)
I use the new stable code to train on all trainval set, and ubmitting on kitti's website with only 15.24 accuracy? It's much low than the paper 16.47 accuracy. Are there any other details to pay attention to when training on the trainval dataset？
